src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private T[] array;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        array = (T[]) new Object[INITIAL_CAPACITY];
+        size = 0;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void grow() {
+        int newCapacity = (int) (array.length * GROWTH_FACTOR);
+        T[] newArray = (T[]) new Object[newCapacity];
+        System.arraycopy(array, 0, newArray, 0, size);
+        array = newArray;
+    }
```

Comment:
Consider renaming the 'grow' method to more accurately reflect the conditional nature of the array growth. For instance, 'growIfArrayFull' would indicate that the array may not always grow when this method is called.